### PR TITLE
Security context

### PR DESCRIPTION
**Pull request by steveperry-53**
The purpose of this PR is to:
* Consolidate Security Context topics.
* Create a step-by-step task where the reader can configure security settings and then verify the results.

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes.github.io/4005)

### CODE REVIEW

**Comment by steveperry-53** [comment]
@Random-Liu, @vvish, @timstclair, The three of you are on my list of tech reviewers for Pods and Containers. @mikedanese, @erictune, I see that you are listed in the edited document.
Would one of you be able to do a tech review for this PR. Thanks.

**Comment by timstclair** [comment]
I know a fair amount about this area. I'm happy to review.

File: docs/tasks/configure-pod-container/security-context.md [hunk 1]
```diff
@@ -0,0 +1,351 @@
...
+
+{% capture overview %}
+
+A security context defines privilege and access control settings for
+a Pod or Container. Security context settings include:
+
+* Discretionary Access Control: Permission to access an object, like a file, is based on
+[user ID (UID) and group ID (GID)](https://wiki.archlinux.org/index.php/users_and_groups).
+
+* [Security Enhanced Linux (SELinux)](https://en.wikipedia.org/wiki/Security-Enhanced_Linux): Objects are assigned security labels.
```
Comment (timstclair) [inline docs/tasks/configure-pod-container/security-context.md]: I feel like it's worth mentioning AppArmor and Seccomp here as well. Technically those are annotations, but they will be moved to fields on the security context soon.

File: docs/tasks/configure-pod-container/security-context.md [hunk 2]
```diff
@@ -0,0 +1,351 @@
...
+See [capability.h](https://github.com/torvalds/linux/blob/master/include/uapi/linux/capability.h)
+for definitions of the capability constants.
+
+**Note**: Linux capability constants have the form `CAP_XXX`. But when you list capabilities
+in your Container manifest, you must omit the `CAP_` portion of the constant. For example,
+to add `CAP_SYS_TIME`, include `SYS_TIME` in your list of capabilities.
+
+## Assign SELinux labels to a Container
+
+To assign SELinux labels to a Container, include the `seLinuxOptions` field in
```
Comment (timstclair) [inline docs/tasks/configure-pod-container/security-context.md]: Note: The host OS must have the SELinux security module loaded in order for SELinux labels to have any affect. If the module is not loaded, I'm not sure whether setting them will give an error, or silently do nothing.

File: docs/tasks/configure-pod-container/security-context.md (hunk 1 above)
Comment (steveperry-53) [inline docs/tasks/configure-pod-container/security-context.md]: Done.

File: docs/tasks/configure-pod-container/security-context.md (hunk 2 above)
Comment (steveperry-53) [inline docs/tasks/configure-pod-container/security-context.md]: Added note.

**Comment by steveperry-53** [comment]
@timstclair, I made some additions to address your comments.

File: docs/tasks/configure-pod-container/security-context.md [hunk 3]
```diff
@@ -23,6 +23,10 @@ a Pod or Container. Security context settings include:
 
 * [Linux Capabilities](https://linux-audit.com/linux-capabilities-hardening-linux-binaries-by-removing-setuid/): Give a process some privileges, but not all the privileges of the root user.
 
+* [AppArmor](https://en.wikipedia.org/wiki/AppArmor): Use program profiles to restrict the capabilities of individual programs.
```
Comment (timstclair) [inline docs/tasks/configure-pod-container/security-context.md]: nit: maybe link to the kubernetes AppArmor docs instead? https://kubernetes.io/docs/tutorials/clusters/apparmor/

Comment (steveperry-53) [inline docs/tasks/configure-pod-container/security-context.md]: Done.

**Comment by steveperry-53** [comment]
@timstclair, I changed one link target according to your suggestion.

**Comment by steveperry-53** [comment]
@timstclair, Thanks for the review!

File: docs/tasks/configure-pod-container/security-context.md [hunk 4]
```diff
@@ -0,0 +1,359 @@
...
+```
+
+In the capability bitmap of the first container, bits 12 and 25 are clear. In the second container,
+bits 12 and 25 are set. Bit 12 is `CAP_NET_ADMIN`, and bit 25 is `CAP_SYS_TIME`.
+See [capability.h](https://github.com/torvalds/linux/blob/master/include/uapi/linux/capability.h)
+for definitions of the capability constants.
+
+**Note**: Linux capability constants have the form `CAP_XXX`. But when you list capabilities
+in your Container manifest, you must omit the `CAP_` portion of the constant. For example,
+to add `CAP_SYS_TIME`, include `SYS_TIME` in your list of capabilities.
```
Comment (MHBauer) [inline docs/tasks/configure-pod-container/security-context.md:304]: Why is this?

**Pull request merged**